firebase-devservices/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/FirebaseDevServiceConfig.java

```diff
@@ -0,0 +1,282 @@
+package io.quarkiverse.googlecloudservices.firebase.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Root configuration class for Google Cloud Firebase that operates at build time.
+ * This class provides a nested structure for configuration, including
+ * a separate group for the development service configuration.
+ * <p>
+ * Here is an example of how to configure these properties:
+ * <p>
+ *
+ * <pre>
+ * quarkus.google.cloud.firebase.devservice.enabled = true
+ * quarkus.google.cloud.pubsub.devservice.image-name = gcr.io/google.com/cloudsdktool/google-cloud-cli # optional
+ * quarkus.google.cloud.pubsub.devservice.emulatorPort = 8085 # optional
+ * </pre>
+ */
+@ConfigMapping(prefix = "quarkus.google.cloud")
+@ConfigRoot
+public interface FirebaseDevServiceConfig {
```

Comment:
Config root must be `quarkus.google.cloud.devservices` so these configs are scoped to the devservices.

The projectId config already exist and should be reused from the common config.

Another solution would be to add the devservices config to each separate module, and if the config is used but the firebase devservices module not included, throw an error at deployment time saying that the firebase devservices module must be included to be able to use pubsub devservcies.

Here there is clearly a dependency between each module that is not correct.
We should take as inspiration how it is done in the AWS service extension pack, however this is a bit different as all devservices are based on localstack, but here some devservices are autonomous and others are based on firebase. https://docs.quarkiverse.io/quarkus-amazon-services/dev/dev-services.html

Reply:
Ok, using a `q.g.c.devservices` config root seems most sensible to me to avoid all the extra dependencies you would get for including the firebase module. Let me work on that.

As for using the project common config interface, please check the issue raised at https://github.com/quarkiverse/quarkus-google-cloud-services/pull/707#discussion_r1882549543

Reply:
Config root has been updated as suggested. Lemme know what you think. Had to rework a bit to get it in line with the changes I did to the testcontainers module, on the bright side, the code is quite a bit cleaner now imho.